stor/tests/test_integration_dx.py

```diff
@@ -0,0 +1,320 @@
+import gzip
+import os
+import pytest
+import time
+import unittest
+from unittest import skipIf
+
+import dxpy
+import mock
+import six
+
+from stor import exceptions
+from stor import NamedTemporaryDirectory
+from stor import Path
+from stor.dx import DXPath
+from stor.test import DXTestCase
+from stor.tests.shared import assert_same_data
+from stor.tests.test_integration import BaseIntegrationTest
+import stor
+import stor.tests.test_integration as test_integration
+
+
+class DXIntegrationTest(BaseIntegrationTest.BaseTestCases, DXTestCase):
+    def setUp(self):
+        self.vcr_enabled = False  # don't want vcr playback for integration tests
+        super(DXIntegrationTest, self).setUp()
+        if not os.environ.get('DX_AUTH_TOKEN'):
+            raise unittest.SkipTest(
+                'DX_AUTH_TOKEN env var not set. Skipping integration test')
+
+        self.setup_temporary_project()
+        self.project = DXPath('dx://' + self.project + ':')
+        self.test_dir = self.project / 'test'
+
+    def tearDown(self):
+        super(DXIntegrationTest, self).tearDown()
+
+    def test_login_auth(self):
+        def mock_header(r, security_context):
+            auth_header = security_context["auth_token_type"] + " " + \
+                          security_context["auth_token"]
+            r.headers[b'Authorization'] = auth_header.encode()
+            return r
+        # dxpy.AUTH_HELPER gets set upon login and called with each api which we mock out here
+        with mock.patch('dxpy.AUTH_HELPER', autospec=True) as mock_auth:
+            mock_auth.security_context = {
+                'auth_token_type': 'Bearer',
+                'auth_token': 'PUBLIC'
+            }
+            mock_auth.side_effect = lambda x: mock_header(x, mock_auth.security_context)
+            with pytest.raises(exceptions.NotFoundError, match='no projects'):
+                self.test_dir.makedirs_p()
+            self.assertEqual(mock_auth.call_count, 1)
+        self.test_dir.makedirs_p()
+        self.assertTrue(self.test_dir.isdir())
+
+    def test_read_bytes_from_binary(self):
```

Comment:
why does this test need to be overwritten?

Reply:
Because of the two lines I added in between. On DX, a file can't be instantaneously read after being written (takes time to go to closed state from closing). `f.wait_on_close(20)` allows for that

Reply:
seems like this could create code bugs, no? perhaps OBSFile.close() should call wait_on_close for dx files?

Also it seems like the amount of time to wait should be configurable and you should catch the timeout error?

Reply:
Yeah, it depends on their platform's load in my experience. f.wait_on_close(20) has never failed for me. f.wait_on_close(10) has passed a few times, failed a few. Either way, we should make it configurable for the tests. I can catch the timeout error, and display a useful message too.

But I feel calling it in OBSFile.close() is not necessary because in some cases, you really don't need to wait for the files to close. e.g. in `test_over_100_files`, we upload about 123 files, and we can't wait 10s for each individual file, instead just choosing to wait for 10s at the very end.

Reply:
Actually the DX error that bubbles up right now says `Reached timeout while waiting for the remote object to close`, which gets raised as a dx.DNAnexusError. I don't think I can word it better than that, so leaving it be.

Reply:
Fixed this, and inherited this test case from super class